pyproject.toml

```diff
@@ -1,5 +1,172 @@
+all = [
+    "h5py", # From wagl (and core)
+    "checksumdir", "netCDF4", # From ancillary
+    "scikit-image", # From algorithms
+    # Add test dependencies from setup.py's tests_require, as test was all
+    "deepdiff>=8.0",
+    "mock",
+    "pep8-naming",
+    "pytest",
+    "rio_cogeo",
+    "sphinx-autodoc-typehints",
+    "sphinx_rtd_theme",
+    "pytest-xdist",
+    "pytest-cov",
+]
+# As per setup.py logic: EXTRAS_REQUIRE["test"] = EXTRAS_REQUIRE["all"]
+test = [
+    "h5py", # From wagl (and core)
+    "checksumdir", "netCDF4", # From ancillary
+    "scikit-image", # From algorithms
+    # Add test dependencies from setup.py's tests_require
+    "deepdiff>=8.0",
+    "mock",
+    "pep8-naming",
+    "pytest",
+    "rio_cogeo",
+    "sphinx-autodoc-typehints",
+    "sphinx_rtd_theme",
+    "pytest-xdist",
+    "pytest-cov",
+]
+# Prod deployment just adds the optional wagl depenencies.
+deployment = ["h5py"] # h5py is already in core, but list it here for clarity if needed.
+
+# --- Setuptools configuration ---
+[tool.setuptools]
+# Let setuptools find packages automatically
+# Equivalent to find_packages(exclude=("tests", "tests.*"))
+# packages = ["eodatasets3"] # Specify main package to include
+
+[tool.setuptools.packages.find]
+# where = ["."] # Search in the current directory
+exclude = ["tests", "tests.*"]
+
+[tool.setuptools.package-data]
+# Equivalent to package_data dictionary
+"*" = ["*.json", "*.yaml", "*.db"]
+"eodatasets3" = ["eodatasets3/py.typed"]
+
+
+# --- setuptools-scm configuration (replaces versioneer) ---
+[tool.setuptools_scm]
+# Where to write the version file
+write_to = "eodatasets3/_version.py"
+# Configure versioning based on git tags
+# Matches tags starting with eodatasets3- (like versioneer's tag_prefix)
+tag_regex = "^eodatasets3-(.*)"
+
+# --- Tool Configurations (from setup.cfg) ---
+
+[tool.pytest]
+testpaths = ["eodatasets3", "tests"] # List of paths
+addopts = "--doctest-modules"
+filterwarnings = ["ignore::FutureWarning"] # List of strings
+norecursedirs = [".*", "build", "dist", ".git", "tmp*"] # List of strings
+
+[tool.flake8]
+# max-complexity = 12 # Commented out as in original
+extend-ignore = [
+    "E203", # See https://github.com/PyCQA/pycodestyle/issues/373
+    "DUO130", # We're stuck with SHA1 for compatibily. Don't warn about it.
+]
+# Use a nested table for per-file-ignores
+per-file-ignores = { "tests/__init__.py" = ["S001"], "tests/test_verify.py" = ["PT009"] }
+max-line-length = 120
+no-accept-encodings = true # boolean
+
+[tool.coverage.run]
+source = ["eodatasets3"] # List of strings
+branch = true # boolean
+
+[tool.coverage.report]
+precision = 2 # integer
+exclude_lines = [
+    "pragma: no cover",
+    "raise NotImplementedError",
+    "raise NotImplemented",
+    "if __name__ == .__main__.:",
+] # List of strings
+
+[tool.isort]
```

Comment:
The precommit configuration only talks about Ruff, should isort be configured at all?

Reply:
Probably not.

At the moment none of `ruff format`, `isort` or `black` are configured to run automatically. Switching to `ruff` for all these makes sense, but is out of scope for this PR.